### PR TITLE
Add env comment

### LINKED ISSUE #2018: sidecar v0.10.0: envsubst does not work
**Thanos, Prometheus and Golang version used**:
v0.10.0

**Object Storage Provider**:
None 

**What happened**:
Trying to render env variable to prometheus config, but it does not work.

prometheus.yml
```yml
global:
  scrape_interval: 15s
  evaluation_interval: 1m
  external_labels:
    location: $LOCATION

```

docker-compose.yml
```yml
version: '2.4'
services:
  thanos-sidecar:
    container_name: thanos-sidecar
    image: thanosio/thanos:v0.10.0
    user: "65534"
    command:
      - sidecar
      - --tsdb.path=/data
      - --prometheus.url=http://prometheus:9090/
      - --reloader.config-envsubst-file=/tmp/prometheus.yml
      - --reloader.config-file=/prometheus.yml
    environment:
      - LOCATION=test
    volumes:
      - ./prometheus.yml:/prometheus.yml
      - /tmp/data:/data
    mem_limit: 150m
    restart: always

  prometheus:
    container_name: prometheus
    hostname: prometheus
    image: prom/prometheus:v2.15.0
    user: "65534"
    command:
      - --web.enable-lifecycle
      - --config.file=/prometheus.yml
      - --storage.tsdb.path=/data
    volumes:
      - ./prometheus.yml:/prometheus.yml
      - /tmp/data:/data
    mem_limit: 1g
    restart: always
```
Check env:
```
$ docker exec -it thanos-sidecar env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=2296288b1629
TERM=xterm
LOCATION=test
HOME=/home
```
Check result:
```
$ docker exec -it thanos-sidecar cat /tmp/prometheus.yml
global:
  scrape_interval: 15s
  evaluation_interval: 1m
  external_labels:
    location: $LOCATION
```

**What you expected to happen**:
Expecting to see rendered config with `$LOCATION` (tried `${LOCATION}` too) expanded to `test`

**Full logs to relevant components**:
```
$ docker logs thanos-sidecar
level=info ts=2020-01-21T11:37:36.7560827Z caller=main.go:149 msg="Tracing will be disabled"
level=info ts=2020-01-21T11:37:36.7561793Z caller=sidecar.go:142 msg="no supported bucket was configured, uploads will be disabled"
level=info ts=2020-01-21T11:37:36.7562848Z caller=options.go:20 protocol=gRPC msg="disabled TLS, key and cert must be set to enable"
level=info ts=2020-01-21T11:37:36.7571157Z caller=sidecar.go:331 msg="starting sidecar"
level=info ts=2020-01-21T11:37:36.7572562Z caller=prober.go:107 msg="changing probe status" status=ready
level=info ts=2020-01-21T11:37:36.7573049Z caller=prober.go:127 msg="changing probe status" status=healthy
level=info ts=2020-01-21T11:37:36.7573505Z caller=http.go:53 service=http/server component=sidecar msg="listening for requests and metrics" address=0.0.0.0:10902
level=info ts=2020-01-21T11:37:36.7578129Z caller=grpc.go:98 service=gRPC/server component=sidecar msg="listening for StoreAPI gRPC" address=0.0.0.0:10901
level=info ts=2020-01-21T11:37:36.760427Z caller=sidecar.go:199 msg="successfully loaded prometheus external labels" external_labels="{location=\"$LOCATION\"}"
level=info ts=2020-01-21T11:37:36.7703093Z caller=reloader.go:286 component=reloader msg="Prometheus reload triggered" cfg_in=/prometheus.yml cfg_out=/tmp/prometheus.yml rule_dirs=
level=info ts=2020-01-21T11:37:36.7705027Z caller=reloader.go:154 component=reloader msg="started watching config file and non-recursively rule dirs for changes" cfg=/prometheus.yml out=/tmp/prometheus.yml dirs=
```

**Anything else we need to know**:
I want to use thanos-sidecar without object storage, just aside with `prometheus` and `git-sync` to have configs updated and prometheus reloaded automatically.

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <764524258@qq.com>

FIx #2018

### CODE REVIEW

File: docs/components/sidecar.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ Prometheus servers connected to the Thanos cluster via the sidecar are subject t
 
   * [Querier](./query.md) to filter out store APIs to touch during query requests.
   * Many object storage readers like [compactor](./compact.md) and [store gateway](./store.md) which groups the blocks by Prometheus source. Each produced TSDB block by Prometheus is labelled with external label by sidecar before upload to object storage.
+  We support use env value to specify it, matching the regular expression `\$\(([a-zA-Z_0-9]+)\)`.
```
Comment (FUSAKLA) [inline docs/components/sidecar.md]: Not sure if we need to overwhelm used with the exact regular expression. 

Maybe just add mention the format to the sentence about env replacing already?
The part in `Reloader` section
```Thanos sidecar can watch `--reloader.config-file=CONFIG_FILE` configuration file,  environment variables found in there, and produce generated config in `--reloader.config-envsubst-file=OUT_CONFIG_FILE` file.```
to
```Thanos sidecar can watch `--reloader.config-file=CONFIG_FILE` configuration file, replace environment variables found in there in `$(VARIABLE)` format, and produce generated config in `--reloader.config-envsubst-file=OUT_CONFIG_FILE` file.```

What do you think?

Comment (daixiang0) [reply docs/components/sidecar.md]: done.

Comment (daixiang0) [reply docs/components/sidecar.md]: @FUSAKLA now promethues support use env in config? Seems i misunderstand.

Comment (FUSAKLA) [reply docs/components/sidecar.md]: No, Prometheus does not support it, this is done by the Thanos sidecar reloader. 
Sorry maybe my formulation is not well understandable.

**Comment by FUSAKLA** [comment]
Thanks @daixiang0 :+1:

**Pull request merged**